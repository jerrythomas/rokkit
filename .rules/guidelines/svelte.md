# Guidelines

- Use Svelte 5 runes mode for code
- Use SvelteKit for app
- Use UnoCSS for styling
- Use vitest for unit testing
- Use Playwright for e2e testing
- E2E tests should follow the page object pattern

## Svelte 5
{{[Svelte & Svelte Kit Distilled](https://svelte-llm.khromov.se/svelte-complete-distilled)}}
{{[svelte 5 documentation](https://svelte.dev/docs/svelte/llms.txt)}}

## Snippets

### Runes

Runes are special functions that replace reactivity declarations from Svelte 4.

```svelte
<script>
  // State
  let count = $state(0);

  // Derived values
  let doubled = $derived(count * 2);

  // Props
  let {
    name = 'world',
    value = $bindable(null)
  } = $props();

  // Effects
  $effect(() => {
    console.log(`Count changed to ${count}`);
  });
</script>
```

### Defining and passing snippets to components:

```svelte
<script>
  let items = ['one','two', 'three']
</script>
<!-- a header snippet -->
{#snippet header()}
  <h1>Custom header</h1>
{/snippet}

<!-- a snippet with parameters -->
{#snippet item(item, selected)}
  <div class:selected={selected}>
    {item.name}
  </div>
{/snippet}

<List {items} {header} {item} />
```

### Using snippets in a component:

```svelte
<script>
  let { header, item } = $props();
</script>

<div class="header">
  {@render header?.()}
</div>

{#each items as currentItem}
  {@render item(currentItem, selectedItem === currentItem)}
{/each}
```

To pass a snippet to another component:

#### ChildComponent.svelte

```svelte
<script>
  let { header } = $props();
</script>

{#if header}
  {@render header()}
{/if}
```

#### Snippets can be passed as properties to child component

```svelte
<script>
   import  ChildComponent from './ChildComponent.svelte'
</script>
{#snippet header()}
  <h1>Custom Header</h1>
{/snippet}

<ChildComponent {header} />
```

### Snippets can be enclosed within the child component.

```svelte
<script>
   import  ChildComponent from './ChildComponent.svelte'
</script>
<ChildComponent>
  {#snippet header()}
    <h1>Custom Header</h1>
  {/snippet}
</ChildComponent>
```
Both of these options work in a similar way

### Important: What NOT to do

Do not confuse snippets with React-style JSX or string templates:

```svelte
<!-- ❌ INCORRECT - Using arrow functions returning strings -->
<List
  header={() => `<h1>My List</h1>`}
  item={({ item }) => `<div>${item.name}</div>`}
/>

<!-- ✅ CORRECT - Using properly defined snippets -->
<!-- snippets are defined in the html section not in scripts block -->
{#snippet header()}
  <h1>My List</h1>
{/snippet}

{#snippet itemTemplate(item, selected)}
  <div class:selected={selected}>
    {item.name}
  </div>
{/snippet}

<List {header} item={itemTemplate} />
```

## Component Design Guidelines

- should be modular and data driven
- should be accessibile
- Reduced motion should be considered

## Project Structure

If the source code uses runes or the test uses runes the file extension should be `.svelte.js` or `.spec.svelte.js` respectively.

```
project/
│
├── src/
│   ├── index.js
│   ├── Component.svelte
│   └── utils.svelte.js
│
├── spec/
│   ├── index.spec.js
│   ├── Component.spec.js
│   └── utils.spec.svelte.js
├── e2e/
│   ├── home.e2e.js
│   └── login.e2e.js
└── README.md
```
