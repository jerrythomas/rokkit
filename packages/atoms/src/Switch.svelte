<script>
	import { equals } from 'ramda'
	import { FieldMapper, noop } from '@rokkit/core'
	import { keyboard } from '@rokkit/actions'
	import Item from './Item.svelte'

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} value
	 * @property {Array<any>} [options]
	 * @property {FieldMapper} [mapping]
	 * @property {boolean} [compact]
	 * @property {boolean} [disabled]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		value = $bindable(),
		options = [false, true],
		fields = new FieldMapper({}, { default: Item }),
		compact = false,
		disabled = false,
		onchange = noop
	} = $props()

	let cursor = $state([])

	function toggle(direction = 1) {
		let nextIndex
		const index = options.indexOf(value)
		if (index === -1) {
			nextIndex = direction === -1 ? options.length - 1 : 0
		} else {
			nextIndex = (index + direction + options.length) % options.length
		}

		value = options[nextIndex]
		onchange(value)
	}

	function handleClick(event) {
		const index = event.target.closest('[data-path]').dataset.path
		value = options[index]
		onchange(value)
	}
	const keyMapping = {
		next: ['ArrowRight', 'ArrowDown', ' ', 'Enter'],
		prev: ['ArrowLeft', 'ArrowUp']
	}
	let useComponent = $derived(!options.every((item) => [false, true].includes(item)))
</script>

{#if !Array.isArray(options) || options.length < 2}
	<rkt-error>Items should be an array with at least two items.</rkt-error>
{:else}
	<!-- svelte-ignore a11y_click_events_have_key_events -->
	<rkt-switch
		class={className}
		class:is-off={options.length === 2 && equals(value, options[0])}
		class:is-on={options.length === 2 && equals(value, options[1])}
		class:compact
		aria-label="Toggle Switch"
		aria-orientation="horizontal"
		aria-disabled={disabled}
		tabindex="0"
		role="listbox"
		use:keyboard={keyMapping}
		onnext={() => toggle()}
		onprev={() => toggle(-1)}
		onclick={handleClick}
	>
		{#each options as item, index (item)}
			{@const SvelteComponent = useComponent ? fields.getComponent(item) : null}
			<rkt-item
				class="relative"
				role="option"
				aria-selected={equals(item, value)}
				data-path={index}
			>
				{#if equals(item, value)}
					<rkt-indicator class="absolute bottom-0 left-0 right-0 top-0"></rkt-indicator>
				{/if}
				{#if SvelteComponent}
					<SvelteComponent value={item} {fields} />
				{/if}
			</rkt-item>
		{/each}
	</rkt-switch>
{/if}
