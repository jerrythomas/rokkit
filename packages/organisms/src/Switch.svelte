<script>
	import { equals } from 'ramda'
	import { defaultFields, getComponent, createEmitter } from '@rokkit/core'
	import { Item } from '@rokkit/molecules'
	import { navigator } from '@rokkit/actions'

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} value
	 * @property {Array<any>} [options]
	 * @property {import('@rokkit/core').FieldMapping} [fields]
	 * @property {any} [using]
	 * @property {boolean} [compact]
	 * @property {boolean} [disabled]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		value = $bindable(),
		options = [false, true],
		fields = $bindable(defaultFields),
		using = $bindable({}),
		compact = false,
		disabled = false,
		...events
	} = $props()

	let cursor = $state([])
	let emitter = createEmitter(events, ['change'])
	function handleNav(event) {
		if (disabled) return
		if (event.detail.node === value) return
		value = event.detail.node
		cursor = event.detail.path
		emitter.change({ item: value, indices: cursor })
	}

	let useComponent = $derived(!options.every((item) => [false, true].includes(item)))
	$effect.pre(() => {
		fields = { ...defaultFields, ...fields }
		using = { default: Item, ...using }
	})
</script>

{#if !Array.isArray(options) || options.length < 2}
	<error>Items should be an array with at least two items.</error>
{:else}
	<toggle-switch
		class="flex items-center {className}"
		class:is-off={options.length === 2 && equals(value, options[0])}
		class:is-on={options.length === 2 && equals(value, options[1])}
		class:compact
		aria-label="Toggle Switch"
		aria-orientation="horizontal"
		aria-disabled={disabled}
		tabindex="0"
		role="listbox"
		use:navigator={{
			items: options,
			fields,
			vertical: false,
			indices: cursor
		}}
		onmove={handleNav}
		onselect={handleNav}
	>
		{#each options as item, index (item)}
			{@const component = useComponent ? getComponent(item, fields, using) : null}
			<item class="relative" role="option" aria-selected={equals(item, value)} data-path={index}>
				{#if equals(item, value)}
					<indicator class="absolute bottom-0 left-0 right-0 top-0"></indicator>
				{/if}
				{#if component}
					{@const SvelteComponent = component}
					<SvelteComponent value={item} {fields} />
				{/if}
			</item>
		{/each}
	</toggle-switch>
{/if}
