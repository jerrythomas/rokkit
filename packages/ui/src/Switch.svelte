<script>
	import { equals } from 'ramda'
	import { noop, FieldMapper } from '@rokkit/core'
	import { keyboard } from '@rokkit/actions'
	import Item from './Item.svelte'
	// import { defaultMapping } from './constants'

	/**
	 * @typedef {Object} Props
	 * @property {string}      [class]
	 * @property {any}         value
	 * @property {Array<any>}  [options]
	 * @property {FieldMapper} [mapping]
	 * @property {boolean}     [compact]
	 * @property {boolean}     [disabled]
	 */

	/** @type {Props} */
	let {
		class: classes = '',
		value = $bindable(),
		options = [false, true],
		fields,
		compact = false,
		disabled = false,
		onchange = noop,
		stub,
		...extra
	} = $props()

	// let cursor = $state([])

	function toggle(direction = 1) {
		let nextIndex
		const index = options.indexOf(value)
		if (index === -1) {
			nextIndex = direction === -1 ? options.length - 1 : 0
		} else {
			nextIndex = (index + direction + options.length) % options.length
		}

		value = options[nextIndex]
		onchange(value)
	}

	function handleClick(event) {
		const index = event.target.closest('[data-path]').dataset.path
		value = options[index]
		onchange(value)
	}
	const keyMapping = {
		next: ['ArrowRight', 'ArrowDown', ' ', 'Enter'],
		prev: ['ArrowLeft', 'ArrowUp']
	}
	// let useComponent = $derived(!options.every((item) => [false, true].includes(item)))
	// let mapper = new FieldMapper(fields)
</script>

{#if !Array.isArray(options) || options.length < 2}
	<rk-error>Items should be an array with at least two items.</rk-error>
{:else}
	<!-- svelte-ignore a11y_click_events_have_key_events -->
	<rk-switch
		class={classes}
		class:is-off={options.length === 2 && equals(value, options[0])}
		class:is-on={options.length === 2 && equals(value, options[1])}
		class:compact
		aria-label="Toggle Switch"
		aria-orientation="horizontal"
		aria-disabled={disabled}
		tabindex="0"
		role="listbox"
		use:keyboard={keyMapping}
		onnext={() => toggle()}
		onprev={() => toggle(-1)}
		onclick={handleClick}
	>
		{#each options as item, index (item)}
			<!-- {@const Template = getSnippet(extra, mapper.get('snippet', item), stub)} -->
			<rk-item class="relative" role="option" aria-selected={equals(item, value)} data-path={index}>
				{#if equals(item, value)}
					<rk-indicator class="absolute bottom-0 left-0 right-0 top-0"></rk-indicator>
				{/if}
				{#if stub}
					{@render stub(item, fields)}
					<!-- <Template value={item} {fields} /> -->
				{:else}
					<Item value={item} {fields} />
				{/if}
			</rk-item>
		{/each}
	</rk-switch>
{/if}
