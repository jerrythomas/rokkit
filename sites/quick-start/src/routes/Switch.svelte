<script>
	import { equals } from 'ramda'
	import { defaultFields, getComponent } from '@rokkit/core'
	import { Item } from '@rokkit/ui'
	import { createView } from '@rokkit/stores'
	import { switchable } from '@rokkit/actions'

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} value
	 * @property {Array<any>} [options]
	 * @property {import('@rokkit/core').FieldMapping} [fields]
	 * @property {any} [using]
	 * @property {boolean} [compact]
	 * @property {boolean} [disabled]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		value = $bindable(false),
		options = [false, true],
		compact = false,
		disabled = false,
		fields = defaultFields,
		using = {}
	} = $props()

	$effect.pre(() => {
		fields = { ...defaultFields, ...fields }
		// using = { default: Item, ...using }
	})

	let actionConfig = $state({
		options,
		value,
		vertical: false,
		index: 0
	})
	const customView = $derived(!options.every((item) => [false, true].includes(item)))
</script>

{#if !Array.isArray(options) || options.length < 2}
	<error>The options should be an array with at least two items.</error>
{:else}
	<toggle-switch
		class="flex items-center {className}"
		class:is-off={options.length === 2 && equals(value, options[0])}
		class:is-on={options.length === 2 && equals(value, options[1])}
		class:compact
		aria-label="Toggle Switch"
		aria-orientation="horizontal"
		aria-disabled={disabled}
		tabindex="0"
		role="listbox"
		use:switchable={actionConfig}
	>
		{#each options as item, index (item)}
			<item class="relative" role="option" aria-selected={equals(item, value)} data-path={index}>
				{#if equals(item, value)}
					<indicator class="absolute bottom-0 left-0 right-0 top-0"></indicator>
				{/if}
				{#if customView}
					{@const SvelteComponent = getComponent(item, fields, using)}
					<SvelteComponent value={item} {fields} />
				{/if}
			</item>
		{/each}
	</toggle-switch>
{/if}
