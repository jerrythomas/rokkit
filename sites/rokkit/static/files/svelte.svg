<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M18.2301 11.1217C21.1087 6.47773 15.3383 1.6935 11.2372 4.37297L7.337 6.85957C5.44834 8.08961 4.62832 10.453 5.78004 12.8817C2.90038 17.2501 8.50365 22.3655 12.7729 19.6332L16.6732 17.1466C18.7952 15.7494 19.2821 13.3398 18.2301 11.1217V11.1217ZM8.46536 8.46761L12.3599 5.98242C14.9184 4.45077 17.8644 7.12896 16.8023 9.62435C16.1715 9.12318 15.4685 8.81503 14.9345 8.67627C15.1375 7.87006 14.2016 7.27202 13.5507 7.68841L9.64763 10.1651C8.68258 10.8321 9.71148 12.204 10.6653 11.6284L12.1527 10.6803C15.4905 8.79676 18.6495 13.3192 15.5448 15.5357L11.6502 18.0209C9.30336 19.5215 6.10922 17.1175 7.20785 14.379C7.79145 14.8207 8.41131 15.1456 9.07564 15.3271C8.92896 16.2531 9.86333 16.7033 10.4694 16.3234L14.3626 13.8382C15.3233 13.1178 14.292 11.814 13.3464 12.375L11.8533 13.3231C8.55362 15.2514 5.32106 10.6843 8.46536 8.46761Z" fill="black"/>
</svg>
