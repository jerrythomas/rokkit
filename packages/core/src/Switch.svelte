<script>
	import { createEventDispatcher } from 'svelte'
	import { defaultFields } from './constants'
	import { Item } from './items'
	import { navigator } from './actions'
	import { getComponent } from './lib'

	const dispatch = createEventDispatcher()

	let className = ''
	export { className as class }
	export let items = [false, true]
	export let fields = defaultFields
	export let using = {}
	export let compact = true
	export let value
	let cursor = []

	function handleNav(event) {
		value = event.detail.node
		cursor = event.detail.path

		dispatch('change', { item: value, indices: cursor })
	}

	$: useComponent = !items.every((item) => [false, true].includes(item))
	$: fields = { ...defaultFields, ...fields }
	$: using = { default: Item, ...using }
</script>

{#if !Array.isArray(items) || items.length < 2}
	<error>Items should be an array with at least two items.</error>
{:else}
	<toggle-switch
		class="flex items-center {className}"
		class:is-off={items.length == 2 && value === items[0]}
		class:is-on={items.length == 2 && value === items[1]}
		class:compact
		tabindex="0"
		role="listbox"
		use:navigator={{
			items,
			fields,
			vertical: false,
			indices: cursor
		}}
		on:move={handleNav}
		on:select={handleNav}
	>
		{#each items as item, index (item)}
			{@const component = useComponent
				? getComponent(item, fields, using)
				: null}
			<item
				class="relative flex"
				role="option"
				aria-selected={item === value}
				data-path={index}
			>
				{#if item == value}
					<indicator class="absolute bottom-0 left-0 right-0 top-0" />
				{/if}
				{#if component}
					<svelte:component this={component} value={item} {fields} />
				{/if}
			</item>
		{/each}
	</toggle-switch>
{/if}
