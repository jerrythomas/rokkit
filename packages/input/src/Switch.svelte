<script>
	import { createEventDispatcher } from 'svelte'

	const dispatch = createEventDispatcher()

	export let label = 'Use Setting'
	export let value = false
	export let tabindex = 0

	function toggle() {
		value = !value
		dispatch('change', { value })
	}
</script>

<button
	role="switch"
	class="flex rounded-full relative"
	{tabindex}
	aria-checked={value}
	class:checked={value}
	on:click={toggle}
>
	<span class="hidden">{label}</span>
	<mark
		aria-hidden="true"
		class="rounded-full absolute"
		class:is-on={value}
		class:is-off={!value}
	/>
</button>

<style>
	.is-on {
		@apply right-1;
	}
	.is-off {
		@apply left-1;
	}
</style>
