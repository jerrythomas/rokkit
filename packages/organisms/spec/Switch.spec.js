import { describe, expect, it, beforeEach } from 'vitest'
import { cleanup, fireEvent, render } from '@testing-library/svelte'
import { tick } from 'svelte'
import Switch from '../src/Switch.svelte'

describe('Switch.svelte', () => {
	beforeEach(() => cleanup())

	it('should render default', async () => {
		const { container } = render(Switch, { value: false })
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
		const items = container.querySelectorAll('toggle-switch item')

		expect(items.length).toBe(2)
		await fireEvent.click(items[1])
		await tick()
		expect(container).toMatchSnapshot()
		await fireEvent.click(items[0])
		await tick()
		expect(container).toMatchSnapshot()
	})
	it('should render with value', () => {
		const { container } = render(Switch, { value: true })
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})
	it('should render text array', async () => {
		const { container, component } = render(Switch, {
			value: null,
			options: ['a', 'b']
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()

		setProperties(component, { options: ['a', 'b', 'c'] })
		await tick()
		expect(container).toMatchSnapshot()
	})
	it('should support editable attribute', () => {
		const { container, component } = render(Switch, {
			value: null,
			options: ['a', 'b']
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()

		setProperties(component, { editable: true })
		expect(container).toMatchSnapshot()
	})
	it('should render using default field mapping', () => {
		const { container } = render(Switch, {
			value: null,
			options: [{ text: 'a' }, { text: 'b' }]
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})
	it('should render using field mappings', () => {
		const { container } = render(Switch, {
			value: null,
			options: [{ name: 'a' }, { name: 'a' }],
			fields: { text: 'name' }
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})
	it('should support custom class', async () => {
		const { container, component } = render(Switch, {
			value: null,
			options: ['a', 'b'],
			class: 'custom'
		})
		let wrapper = container.querySelector('toggle-switch')
		expect(Array.from(wrapper.classList)).toContain('custom')

		setProperties(component, { class: 'other' })
		await tick()
		wrapper = container.querySelector('toggle-switch')
		expect(Array.from(wrapper.classList)).toContain('other')
	})
	it('should support custom icons', async () => {
		const { container, component } = render(Switch, {
			value: 'a',
			options: ['a', 'b']
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
		setProperties(component, { icons: { remove: 'close' } })
		await tick()
		expect(container).toMatchSnapshot()
	})
	// it('should render items using custom component', () => {})
	// it('should handle item value changes', () => {})
	// it('should pass select and change events', () => {})
	// it('should render items using custom component', () => {})
	// it('should expand and collapse', () => {})
	// it('should pass select and change events', () => {})
})
