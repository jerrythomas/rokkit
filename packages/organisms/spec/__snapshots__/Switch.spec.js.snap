// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch.svelte > should render default 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center is-off"
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="true"
        class="relative"
        data-path="0"
        role="option"
      >
        <indicator
          class="absolute bottom-0 left-0 right-0 top-0"
        />
         
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should render text array 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center "
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="false"
        class="relative"
        data-path="0"
        role="option"
      >
         
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
         
        <p>
          b
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should render using default field mapping 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center "
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="false"
        class="relative"
        data-path="0"
        role="option"
      >
         
        
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
        
         
        <p>
          b
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should render using field mappings 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center "
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="false"
        class="relative"
        data-path="0"
        role="option"
      >
         
        
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
        
         
        <p>
          a
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should render with value 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center is-on"
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="false"
        class="relative"
        data-path="0"
        role="option"
      >
         
         
      </item>
      <item
        aria-selected="true"
        class="relative"
        data-path="1"
        role="option"
      >
        <indicator
          class="absolute bottom-0 left-0 right-0 top-0"
        />
         
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should support custom icons 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center is-off"
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="true"
        class="relative"
        data-path="0"
        role="option"
      >
        <indicator
          class="absolute bottom-0 left-0 right-0 top-0"
        />
         
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
         
        <p>
          b
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should support custom icons 2`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center is-off"
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="true"
        class="relative"
        data-path="0"
        role="option"
      >
        <indicator
          class="absolute bottom-0 left-0 right-0 top-0"
        />
         
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
         
        <p>
          b
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should support editable attribute 1`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center "
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="false"
        class="relative"
        data-path="0"
        role="option"
      >
         
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
         
        <p>
          b
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;

exports[`Switch.svelte > should support editable attribute 2`] = `
<body>
  <div>
    <toggle-switch
      class="flex items-center "
      role="listbox"
      tabindex="0"
    >
      <item
        aria-selected="false"
        class="relative"
        data-path="0"
        role="option"
      >
         
         
        <p>
          a
        </p>
        
        
         
      </item>
      <item
        aria-selected="false"
        class="relative"
        data-path="1"
        role="option"
      >
         
         
        <p>
          b
        </p>
        
        
         
      </item>
    </toggle-switch>
    
  </div>
</body>
`;
